modules/ingester/instance.go

```diff
@@ -174,35 +174,44 @@ func (i *instance) CompleteBlock(blockID uuid.UUID) (uuid.UUID, error) {
 	i.blocksMtx.Unlock()
 
 	if completingBlock == nil {
-		return uuid.Nil, fmt.Errorf("error finding completingBlock")
+		return fmt.Errorf("error finding completingBlock")
 	}
 
 	// potentially long running operation placed outside blocksMtx
 	completeBlock, err := i.writer.CompleteBlock(completingBlock, i)
 	if err != nil {
 		metricFailedFlushes.Inc()
 		level.Error(log.Logger).Log("msg", "unable to complete block.", "tenantID", i.instanceID, "err", err)
-		return uuid.Nil, err
+		return err
 	}
 
-	// remove completingBlock and add completeBlock
 	i.blocksMtx.Lock()
+	i.completeBlocks = append(i.completeBlocks, completeBlock)
+	i.blocksMtx.Unlock()
+
+	return nil
+}
+
+func (i *instance) ClearCompletingBlock(blockID uuid.UUID) {
+	i.blocksMtx.Lock()
+	var completingBlock *wal.AppendBlock
 	for j, iterBlock := range i.completingBlocks {
 		if iterBlock.BlockID() == blockID {
+			completingBlock = iterBlock
 			i.completingBlocks = append(i.completingBlocks[:j], i.completingBlocks[j+1:]...)
 			break
 		}
 	}
-	completeBlockID := completeBlock.BlockMeta().BlockID
-	i.completeBlocks = append(i.completeBlocks, completeBlock)
 	i.blocksMtx.Unlock()
 
-	err = completingBlock.Clear()
-	if err != nil {
-		level.Error(log.Logger).Log("msg", "Error clearing wal", "tenantID", i.instanceID, "err", err)
+	if completingBlock != nil {
+		err := completingBlock.Clear()
+		if err != nil {
+			level.Error(log.Logger).Log("msg", "Error clearing wal", "tenantID", i.instanceID, "blockID", blockID.String(), "err", err)
```

Comment:
should we metric this somewhere? if we're unable to delete files for some reason and they start piling up would be nice to have some visibility.

maybe return err from this method and increment failed flushes in the calling code? (but don't actually fail the flush, just log and metric)